src/EnergyPlus/ScheduleManager.cc

```diff
@@ -2877,6 +2878,12 @@ namespace ScheduleManager {
             thisHolidayIndex = DataEnvironment::HolidayIndexTomorrow;
         }
 
+        // In the case where DST is applied on 12/31 at 24:00, which is the case for a Southern Hemisphere location for eg
+        // (DayOfYear_Schedule is a bit weird, ScheduleManager always assumes LeapYear)
+        if (thisDayOfYear == 367) {
+            thisDayOfYear = DataEnvironment::RunPeriodStartJDay;
+        }
```

Comment:
Optional second way of doing it... I'm just wary of cases where you might have used some kind of Schedule that doesn't have entries for all days, such as using a Schedule:Year with Schedule:Week:Daly and Schedule:Day: it's possible that January 1 isn't set...

Though by definition, the day after December 31 should be January 1... so the first way should be more correct.

Reply:
https://github.com/NREL/EnergyPlus/commit/fcad093296b8ffe56a09c18825a12a1fefbe1b72 should probably be reverted

Reply:
@rraustad @mjwitte @Myoldmopar any thoughts on this comment and the one just above please?

Reply:
That's an interesting logic problem. What is the next day of the simulation after Dec 31? Yeah, It's always Jan 1, if Jan 1 is in the simulation. The RunPeriod could be 2/1 - 12/31 or 4/1 - 3/30. I guess you'd only need that if block using my second RunPeriod dates example.

Reply:
Seems like it should use Jan 1.  EnergyPlus expects schedules to be fully defined for the entire year and warns if they are not (at least that was the case long ago, I assume it still is). Perhaps a note in the I/O ref? "If DST is active on Dec 31, then hour 24 will use the schedule values for Jan 1, hour 1."

Reply:
You're right, I just tried ahving a Schedule:Year with missing days:

```
Program Version,EnergyPlus, Version 9.4.0-db88b719ab, YMD=2020.06.30 18:02,
   ** Severe  ** ProcessScheduleInput: Schedule:Year="SOUTHZONE-CONTROL SCHED" has missing days in its schedule pointers
   ** Severe  ** ProcessScheduleInput: Schedule:Year="SOUTHZONE-SETPOINTS" has missing days in its schedule pointers
   **  Fatal  ** ProcessScheduleInput: Preceding Errors cause termination.

```

I'll revert fcad093 then